app/src/main/java/tech/relaycorp/courier/data/network/Message.kt

```diff
@@ -0,0 +1,17 @@
+package tech.relaycorp.courier.data.network
+
+import java.io.InputStream
+import java.util.Date
+
+// Mock RAMFMessage to help us with the high-level implementation
+class Message(
+    val recipientAddress: String,
+    val messageId: String,
+    val creationTime: Date,
+    val ttl: Int,
+    val payload: ByteArray
+) {
+    companion object {
+        fun wrap(data: InputStream) = Message("", "", Date(), 0, ByteArray(0))
+    }
+}
```

Comment:
This is already implemented in the core JVM library, although it's an abstract class and I haven't created the two subclasses you'll use here: `Cargo` and `CargoCollectionAuthorization`. The serialization and deserialization methods are already implemented, along with these fields, so implementing those subclasses should be easy -- should I do it now?

Reply:
We can do it now, or keep it stubbed for a while. I just wanted to grasp the general architecture of things.

Reply:
Cool, in that case let's keep it until the the missing bits are implemented in the JVM lib.